### DISCUSSION TITLE
kiali warning Warning alert:failed to get mesh version, please check if url_service_version is configured correctly.

### DISCUSSION THREAD

**Discussion by ryadama9** [kiali/kiali#6247] [Q&A]
we are running kiali 1.50 version. I an currently seeing a warning in kiali UI saying Warning alert:failed to get mesh version, please check if url_service_version is configured correctly. we have installed kiali using helm operator in istio-system namespace and kiali-operator is running on kiali-operator namespace. I tried passing url_service_version to kiali helm chart. It didn't help to resolve this warning. Did anyone faced similar issue and resolved it? Also, any thoughts on troubleshooting this. Thank you

<img width="473" alt="Screenshot 2023-06-07 at 1 54 01 PM" src="https://github.com/kiali/kiali/assets/102109599/660e98cc-f1fc-403f-a30a-2950612443f9">

**Comment by jmazzitelli** [comment]
_converted to Q&A discussion_

If you look in the kiali server pod logs, do you see anything that could tell you why the version couldn't be determined? If nothing in logs, turn on trace logging, maybe more information will be logged.

The default is defined here: https://github.com/kiali/kiali-operator/blob/e5a7af702e776c8dd1df8a1a256af21cd43e5295/roles/default/kiali-deploy/tasks/main.yml#L229

`'http://' + kiali_vars.external_services.istio.istiod_deployment_name + '.' + kiali_vars.istio_namespace + ':15014/version`

So it needs to be the istiod endpoint that provides the version information. If you did anything to affect that default (changed the name of the istiod deployment, such as using canary deployments, or you changed/disabled its port 15014) you will get this error.

**Reply by ryadama9** [reply]
yes, you are right. Thank you. I updated url_service_version according to istiod deployment and now it throws a different error saying Kiali [v1.50.0] may not be compatible with [Istio 1.14.5], and is not recommended. See kiali.io for version compatibility. when I looked at compatibility matrix I see 1.14 Istio is compatible from 1.50 kiali [here](https://kiali.io/docs/installation/installation-guide/prerequisites/)

**Reply by jmazzitelli** [reply]
Kiali got rid of that version compatibility check code in a more recent version - it was getting outdated and hard to maintain (as you see here). Ignore that error if the documentation says it is compatible. When you upgrade to a newer Kiali, those compatibility checks at run time are no longer there.

**Reply by ryadama9** [reply]
sure, Thank you 👍

**Reply by jshaughn** [reply]
Accordingtto the chart, you could update to 1.54 of Kiali and stay compatible.

**Reply by ryadama9** [reply]
Sure, Thank you. @jshaughn
